### DISCUSSION TITLE
KeyCount attribute for BeatmapAttributeText?

### DISCUSSION THREAD

**Discussion by LittleLilyBun** [ppy/osu#27537] [Ideas]
Title is pretty self-explanatory, I think having a KeyCount attribute for BeatmapAttributeText in the skin editor could be useful in mania skins. Though, I could see there being issues with how it would display on song select when switching between modes, or when put into a non-mania skin layout.

**Comment by Stoppedpuma** [comment]
Does (Default/Argon)KeyCounterDisplay not do what you're asking? (Default/Argon)KeyCounterDisplay keeps track of how many times you press a key, do you mean an element which shows how many notes are remaining or are in the map?

**Reply by nagi-desuuu** [reply]
I think they're asking for the Key-Count in a map, like 7K or 4K.

**Reply by LittleLilyBun** [reply]
Yeah, I'm looking for the mania key counts like 4K and 7K

**Reply by Stoppedpuma** [reply]
I guess I don't fully understand a reason for this besides very high key counts like 14 where it's a bit hard to tell by just glancing down in gameplay? You mention song select, does the one on song select not do this well enough?
![1710133820](https://github.com/ppy/osu/assets/58333920/bc90f6ef-45d1-4a7d-a88b-2712b43ddb32)

**Reply by LittleLilyBun** [reply]
Sorry if I'm not very clear, I was mentioning song select because there may be issues with how a KeyCount attribute would show up when on song select and on a different mode from mania, since you're able to place BeatmapAttributeText on song select.

**Reply by Stoppedpuma** [reply]
> mentioning song select because there may be issues with how a KeyCount attribute would show

I don't think there would be an issue displaying it on song select since the Key Count in the image above works fine.

> when on song select and on a different mode from mania

I don't see how the KeyCount attribute would be useful for modes besides mania since they don't have various key counts. Maybe this could lead to per gamemode attributes such as "FruitsCaught" for catch or "CirclesClicked" for osu! etc?
